example/debug/odp_debug.c

```diff
@@ -382,12 +383,19 @@ static int timer_debug(void)
 	if (timer_capa.max_tmo.max_tmo < max_tmo)
 		max_tmo = timer_capa.max_tmo.max_tmo;
 
-	if (timer_capa.max_tmo.res_ns > res)
-		res = timer_capa.max_tmo.res_ns;
+	memset(&timer_res_capa, 0, sizeof(odp_timer_res_capability_t));
+	timer_res_capa.max_tmo = max_tmo;
+	if (odp_timer_res_capability(ODP_CLOCK_CPU, &timer_res_capa)) {
+		ODPH_ERR("Timer resolution capability failed\n");
+		return -1;
+	}
+
+	if (timer_res_capa.res_ns > res)
+		res = timer_res_capa.res_ns;
 
 	memset(&timer_param, 0, sizeof(timer_param));
 	timer_param.res_ns  = res;
-	timer_param.min_tmo = max_tmo / 10;
+	timer_param.min_tmo = res;
```

Comment:
I think you had earlier timer_res_capa.min_tmo here. Value of res may cause problems here as timer_res_capa.min_tmo value may be larger than timer_res_capa.res_ns. So, it would be safer to use timer_res_capa.min_tmo, or something around max_tmo (but less than max_tmo/2 which we are using for timeouts).

Reply:
Ping @PavanNikhilesh